skema/skema_py/server.py

```diff
@@ -65,7 +71,38 @@ class System(BaseModel):
     )
 
 
-def system_to_gromet(system: System):
+async def system_to_enriched_system(system: System) -> System:
+    """Takes a System as input and enriches it with comments by running the Rust comment extractor."""
+
+    def get_language_from_extension(extension: str) -> str:
+        """Function for converting file extension to language enum used by CodeSnippet"""
+        extension = extension.lower()
+        extension_map = {
+            ".f": "Fortran",
+            ".f95": "Fortran",
+            ".for": "Fortran",
+            ".py": "Python",
+            ".cpp": "CppOrC",
+            ".c": "CppOrC",
+        }
+        return extension_map.get(extension, "")
+
+    comments = {"files": {}}
+    for file, blob in zip(system.files, system.blobs):
+        file_path = Path(system.root_name or "") / file
+
+        snippet = CodeSnippet(
+            code=blob, language=get_language_from_extension(file_path.suffix)
+        )
+        result = await comments_proxy.proxy_extract_comments(snippet)
```

Comment:
to avoid blocking, we might want to try using `await asyncio.gather()`.

Reply:
I've now updated this part of the code to use asyncio.gather() instead.